Sources/NIOCore/AsyncChannel/AsyncChannel.swift

```diff
@@ -301,15 +301,13 @@ public struct NIOAsyncChannel<Inbound: Sendable, Outbound: Sendable>: Sendable {
             }
         }
 
-        do {
-            self._outbound.finish()
-            try await self.channel.close().get()
-        } catch {
-            if let error = error as? ChannelError, error == .alreadyClosed {
-                return result
-            }
-            throw error
-        }
+        self._outbound.finish()
+        // We ignore errors from close, since all we care about is that the channel has been closed
+        // at this point.
+        self.channel.close(promise: nil)
+        // `closeFuture` is never failed, so we can ignore the error
+        try? await self.channel.closeFuture.get()
```

Comment:
Not sure whether we decided to ignore `closeFuture` errors entirely or not – it seems completely reasonable to ignore them though. 

Suspect @FranzBusch has opinions.

Reply:
This future should never be failed; if we're considering a failing `closeFuture` the result of a bad implementation of a `Channel` but not something that should happen in any other scenario, it seemed reasonable to me to ignore all errors.

Reply:
The original goals were:
1. Make sure we only excited from `executeThenClose` when the channel is actually closed
2. Surface any errors if closing fails

Now 1. is still being upheld here since we wait for the `closeFuture`. It also turns out that closing can't fail and a call to `close()` must always result in the closeFuture completing. Strictly speaking succeeding actually.

Now we have two options here. We can just assume that the "strict" part is true so there will never be any errors. However, there might be some code out there that actually fails the close future. What if we catch the error and add an assert that triggers if this close future actually fails to help us spot incorrect implementations?

Reply:
Fair enough, adding an assertion sounds useful.